app/controllers/admin/users_controller.rb

```diff
@@ -191,8 +191,26 @@ def revoke_admin
   end
 
   def grant_admin
-    AdminConfirmation.new(@user, current_user).create_confirmation
-    render json: success_json
+    if current_user.has_any_second_factor_methods_enabled?
```

Comment:
This should probably be moved to ` AdminConfirmation`  that way we can be sure to have both workflow in the same place.

I think we've bypassing ` guardian.ensure_can_grant_admin!(@target_user)`  with 2FA, which we don't want to.